### ISSUE TITLE
TextField TextInputAction.newline doesn't work properly on Android

### ISSUE THREAD

**Issue by njovy** [flutter/flutter#54745]
## Steps to Reproduce

```dart
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return new MaterialApp(
      home: new HomePage(),
    );
  }
}

class HomePage extends StatefulWidget {
  @override
  HomePageState createState() => HomePageState();
}

class HomePageState extends State<HomePage> {
  TextEditingController controller1 = TextEditingController();
  FocusNode focusNode1 = FocusNode();

  @override
  void initState() {
    super.initState();
  }

  @override
  void dispose() {
    controller1.dispose();
    focusNode1.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: Text("Multiline Text Field"),),
      body: SafeArea(
        child: SingleChildScrollView(
          child: Column(
            children: <Widget>[
              TextField(
                controller: controller1,
                focusNode: focusNode1,
                maxLines: 10,
                minLines: 7,
                maxLength: 200,
                textInputAction: TextInputAction.newline,

                keyboardType: TextInputType.text,
              ),
            ],
          ),
        ),
      ),
    );
  }
}
```
@GaryQian  Above is a code that you can reproduce this bug.

Press the return key doesn't insert a line feed properly.
1. Samsung Keyboard: It doesn't insert a new line at all.
2. Gboard: It only insert a new line once even if you press the return key multiple times.

## Logs
```
[✓] Flutter (Channel master, v1.18.0-6.0.pre.19, on Mac OS X 10.15.3 19D76, locale en-KR)
    • Flutter version 1.18.0-6.0.pre.19 at /Users/captainpark/Library/Flutter/sdk
    • Framework revision 9cc69d47a5 (10 hours ago), 2020-04-13 19:00:02 -0700
    • Engine revision deef2663ac
    • Dart version 2.8.0 (build 2.8.0-dev.20.0 c9710e5059)

  • SM G965N                             • android-arm64 • Android 9 (API 28)

```
![newline](https://user-images.githubusercontent.com/1792070/80252231-dfc84980-86b1-11ea-9963-770d3596511e.gif)

**Comment by VladyslavBondarenko** [comment]
Hi @njovy 
I tried with `master v1.18.0-6.0.pre.19` and `dev v1.18.0-dev.5.0` with Redmi Note 7 (Android 9). Newline is working as intended with Gboard and with Faceimoji Keyboard for Xiaomi.

With what devices are you experiencing this?

**Comment by njovy** [comment]
@VladyslavBondarenko I tested this on Galaxy S9+

**Comment by njovy** [comment]
@VladyslavBondarenko @GaryQian The issue appears in the latest beta as well. I tested on my nexus 5x and it only inserts a new line once even if you press the enter key multiple times. Seems like this needs to be fixed asap.

**Comment by njovy** [comment]
One more edit: The issue only appears with Korean input.

**Comment by njovy** [comment]
![newline](https://user-images.githubusercontent.com/1792070/80252182-c921f280-86b1-11ea-9303-2fc0d8581a6d.gif)

**Comment by GaryQian** [comment]
Did some experimentation, this occurs when the InputConnectionAdaptor's sendKeyEvent is triggered instead of simply adding a \n character. This seems to be IME dependent, as some keyboards choose to make the newline button a key event while others choose to just append a \n character.

Our code behaves such that the newline key event performs no additional work since we are assuming that the IME has already inserted the \n. This is evidently not true for some keyboards, which only call sendKeyEvent without inserting the character.

**Comment by njovy** [comment]
@GaryQian I set onSubmitted callback to see if I can add \n at the end of an input as a workaround but it never gets called.
After a few different tries, I realized that I specified a keyboardType with TextInputType.text for a multiline textfield. I switched to TextInputType.multiline and the issue went away. As setting a wrong keyboard type can cause an unexpected behavior, add one more assertion to avoid this would be suffice. Thank you always!
